// Copyright 2018 Google LLC. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Configuration for segment normalizer.

syntax = "proto2";

package fonbund;

// Next available ID: 3
message SegmentNormalizerConfig {
  // Symbol rewrites. A basic regular expression with LHR and RHS parts, which
  // can be restricted to particular languages.
  //
  // Next available ID: 4
  message SymbolRewrite {
    // Source segment.
    optional string lhs = 1;

    // Target segment the source is rewritten to.
    optional string rhs = 2;

    // A list of language/region pairs in BCP-47 format (e.g. "en-IN")
    // to which this rewrite is restricted to. If empty, the rewrite
    // applies to all the languages. Otherwise, the rewrite will only apply if
    // the language is in the list.
    repeated string restrict_to = 3;
  };

  // Remove ligature ties and underties.
  optional bool remove_tie_bars = 1;

  // List of symbol rewrites.
  repeated SymbolRewrite symbol_rewrites = 2;
}
